File: CURIOSITY/bootcamp-research/_cspence/Etzy-BackEnd/db/models/favorite.js
'use strict';
module.exports = (sequelize, DataTypes) => {
  const Favorite = sequelize.define('Favorite', {
    userId: {
      allowNull: false,
      references: { model: 'Users' },
      type: DataTypes.INTEGER
    },
    favProduct: {
      allowNull: false,
      type: DataTypes.BOOLEAN
    },
    shopId: {
      defaultValue: null,
      references: { model: 'Products' },
      type: DataTypes.INTEGER
    },
    productId: {
      defaultValue: null,
      references: { model: 'Products' },
      type: DataTypes.INTEGER
    }
  }, {});
  Favorite.associate = function (models) {
    Favorite.belongsTo(models.Product, { foreignKey: 'productId' });
    Favorite.belongsTo(models.User, { foreignKey: 'userId' });
    Favorite.belongsTo(models.Shop, { foreignKey: 'shopId' });
  };
  return Favorite;
};

File: bootcamp-research/_cspence/CoffeeSuppliers/app/models/favorite.py
from .db import db


class Favorite(db.Model):
    __tablename__ = 'favorites'

    id = db.Column(db.Integer, primary_key=True)
    user_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False)
    product_id = db.Column(db.Integer, db.ForeignKey(
        'products.id'), nullable=False)

    def to_dict(self):
        return {
            "id": self.id,
            "userId": self.user_id,
            "productId": self.product_id
        }
